packages/amplify-category-function/src/provider-utils/awscloudformation/utils/lambda-layer-cloudformation-template.ts

```diff
@@ -30,7 +30,12 @@ export function generateLayerCfnObj(isNewVersion: boolean, parameters: LayerPara
     layerCloudState.latestVersionLogicalId = logicalName;
     versionList.unshift({ LogicalName: logicalName, legacyLayer: false });
   } else {
-    logicalName = _.first(versionList).LogicalName;
+    logicalName = _.first(versionList)?.LogicalName;
+    if (!logicalName) {
+      throw new AmplifyError('LambdaLayerNotFoundError', {
+        message: 'No versions were found for the Lambda Layer. Were they deleted?',
```

Comment:
if this refers to layers stored in the cloud we should somehow convey where the deletion could have happen.
I'm assuming the deletion must have happened outside of amplify - should that be part of the question?

Reply:
Tried to make the message a bit clearer. I was not able to reproduce the specific error, so I can't give much better guidance than to check the console at this point. I'm open to suggestions on messaging.